apps/web-connect/src/features/checkout/WebBuyKeysOrderTotal.tsx

```diff
@@ -66,6 +74,65 @@ export function WebBuyKeysOrderTotal(
 		}
 	};
 
+	const handleCurrencyChange = async (newCurrency:	string) => {
+		setCurrency(newCurrency);
+		const walletAddress = address ? address : "";
+		if(newCurrency === "AETH") {
+			setTokenBalance(0n);
+			setTokenAllowance(0n);
+		}else if(newCurrency === "XAI") {
+			const xaiBalance = await getXaiBalance(walletAddress);
+			const xaiAllowance = await getXaiAllowance(walletAddress, config.nodeLicenseAddress);
+			setTokenBalance(xaiBalance.balance);
+			setTokenAllowance(xaiAllowance.approvalAmount);
+		}else{
+			const esxaiBalance = await getXaiBalance(walletAddress); 
```

Comment:
It should get the esXaiBalance

Reply:
Great catch, this has been fixed. Pushing a new PR momentarily.